### PR TITLE
BXMSPROD-8: refactor to build without jboss-ip-bom

### PR DESCRIPTION
**Pull request by mbiarnes**
fixing uberfire-packaging submodules and duplicate dependency

removed duplicated dependency

added missing deps because of enforcer-rule

disabled maven-jar-plugin

added missing deps

change type of artifact to pom

missing deps adn excludes for testing

updated kie-parent to 7.22.0-SNAPSHOT

updating missing versions to 7.23.0.-SNAPSHOT

disbled maven-jar-plugin

BXMSPROD-333: implementing new methods from javax.enterprise.event.Event

(cherry picked from commit 25899f41bbca0da5923e669bda2ab91c2d28cd61)

upgraded missing deps to 7.24.0-SNAPSHOT

### CODE REVIEW

**Comment by mbiarnes** [comment]
https://github.com/kiegroup/kie-soup/pull/95
https://github.com/kiegroup/appformer/pull/718
https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/973
https://github.com/kiegroup/droolsjbpm-knowledge/pull/391
https://github.com/kiegroup/drools/pull/2396
https://github.com/kiegroup/optaplanner/pull/514
https://github.com/kiegroup/jbpm/pull/1503
https://github.com/kiegroup/kie-jpmml-integration/pull/5
https://github.com/kiegroup/droolsjbpm-integration/pull/1849
https://github.com/kiegroup/droolsjbpm-tools/pull/95
https://github.com/kiegroup/kie-uberfire-extensions/pull/85
https://github.com/kiegroup/kie-wb-playground/pull/76
https://github.com/kiegroup/kie-wb-common/pull/2710
https://github.com/kiegroup/drools-wb/pull/1169
https://github.com/kiegroup/optaplanner-wb/pull/339
https://github.com/kiegroup/jbpm-designer/pull/854
https://github.com/kiegroup/jbpm-wb/pull/1364
https://github.com/kiegroup/kie-docs/pull/1645
https://github.com/kiegroup/kie-wb-distributions/pull/940
https://github.com/kiegroup/optaweb-employee-rostering/pull/250

File: pom.xml [hunk 1]
```diff
@@ -885,6 +849,14 @@
...
   <dependencyManagement>
     <dependencies>
+      <dependency>
+        <groupId>org.kie</groupId>
+        <artifactId>kie-platform-bom</artifactId>
+	    <version>${version.org.kie}</version>
+	    <type>pom</type>
+	    <scope>import</scope>
+      </dependency> 	    
+
```
Comment (tiagobento) [inline pom.xml]: Check indentation here.

**Comment by mbiarnes** [comment]
@ederign @tiagobento please check the failed tests in the Linux build

**Comment by mareknovotny** [comment]
Something wrong with empty beans.xml parsing by weld? 
`javax.enterprise:cdi-api:jar:1.2` vs javax.enterprise:cdi-api:jar:2.0.SP1 ?

```
Error parsing jar:file:/home/jenkins/workspace/KIE/master/pullrequest/appformer-pullrequests/uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-api/target/uberfire-commons-editor-api-2.21.0-SNAPSHOT.jar!/META-INF/beans.xml
```

**Comment by mbiarnes** [comment]
Jenkins please retest this

**Pull request merged**